app/main.py

```
@@ -1,6 +1,31 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Any
+
+
+def is_inmutable(value: Any) -> bool:
+    return isinstance(value, (int, float, complex, str,
+                              tuple, frozenset, bytes))
+    """Check if argument is inmutable."""
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    store = {}  # Contains cached results
+
+    @wraps(func)
+    def inner(*args, **kwargs) -> Any:
+
+        key = args + tuple(sorted(kwargs.items()))
+
+        if not all(is_inmutable(value)for value in key):
+            print("Argument is mutable, won't be cached")
+            return None
```

Printing `"Argument is mutable, won't be cached"` and returning `None` stops the decorated function from executing for mutable arguments. The checklist instructs to remove debug prints before committing: **"Add comments, prints, and functions to check your solution when you write your code. Don't forget to delete them when you are ready to commit and push your code."**  Instead of returning `None`, either (a) call the original function and return its result without caching, or (b) raise a clear error. Do not silently return `None` for non-immutable inputs.